src/matlab2tikz.m

```diff
@@ -3692,19 +3694,20 @@ case size(cData,2) % specific y-data is given
     % define arrow style
     arrowOptions = opts_print(m2t, arrowOpts, ',');
 
-    % Append the arrow style to the TikZ options themselves.
-    % TODO: Look into replacing this by something more 'local',
-    % (see \pgfplotset{}).
     m2t.content.options = opts_add(m2t.content.options,...
         sprintf('arrow%d/.style', m2t.quiverId), ...
         ['{', arrowOptions, '}']);
-    % - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    % return the vector field code
-    str = [str, ...
-        sprintf(['\\',name,' [arrow',num2str(m2t.quiverId), '] ', ...
-        'coordinates{(',format,') (',format,')};\n'],...
-        data)];
-    %FIXME: external
+    
+    tabOpts = 'row sep=crcr';
+    [m2t, table] = makeTable(m2t, repmat({''}, size(data,2)), data);
+    str = sprintf(['\\%s[%%\npoint meta={sqrt((\\thisrow{u})^2+(\\thisrow{v})^2)},\n'...
```

Comment:
Most of these things only account for 2D arrows, so we should also account for 3D.